nncf/automl/environment/quantization_env.py

```diff
@@ -0,0 +1,612 @@
+import logging
+
+import os.path as osp
+import sys
+import time
+from pathlib import Path
+
+import torch
+import torch.backends.cudnn as cudnn
+import torch.nn as nn
+import torch.nn.parallel
+import torch.optim
+import torch.utils.data
+import torch.utils.data.distributed
+import torchvision.datasets as datasets
+import torchvision.models as models
+import torchvision.transforms as transforms
+import warnings
+
+import json
+import math
+import numpy as np
+import pandas as pd
+from copy import deepcopy
+from natsort import natsorted
+from collections import OrderedDict
+
+from functools import partial
+from shutil import copyfile
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+from torchvision.datasets import CIFAR10, CIFAR100
+
+from nncf.nncf_logger import logger
+
+from nncf import create_compressed_model
+from nncf.compression_method_api import CompressionLevel
+from nncf.dynamic_graph.graph_builder import create_input_infos
+from nncf.initialization import register_default_init_args
+from sklearn.preprocessing import MinMaxScaler
+
+from nncf.automl.utils.utils import AverageMeter, topk_accuracy, annotate_model_attr
+
+from collections import OrderedDict, Counter
+from nncf.quantization.quantizer_id import WeightQuantizerId, NonWeightQuantizerId, InputQuantizerId, FunctionQuantizerId
+from nncf.dynamic_graph.context import Scope
+from natsort import natsorted
+
+from nncf.quantization.precision_init.adjacent_quantizers import GroupsOfAdjacentQuantizers, AdjacentQuantizers
+from nncf.hw_config import HWConfigType
+from nncf.quantization.layers import BaseQuantizer
+
+# logging
+def prRed(prt): logger.info("\033[91m {}\033[00m" .format(prt))
+def prGreen(prt): logger.info("\033[92m {}\033[00m" .format(prt))
+def prYellow(prt): logger.info("\033[93m {}\033[00m" .format(prt))
+def prLightPurple(prt): logger.info("\033[94m {}\033[00m" .format(prt))
+def prPurple(prt): logger.info("\033[95m {}\033[00m" .format(prt))
+def prCyan(prt): logger.info("\033[96m {}\033[00m" .format(prt))
+def prLightGray(prt): logger.info("\033[97m {}\033[00m" .format(prt))
+def prBlack(prt): logger.info("\033[98m {}\033[00m" .format(prt))
+
+def find_qid_by_str(quantization_controller, qid_str):
+    for _qid, _q in quantization_controller.all_quantizations.items():
+        if qid_str == str(_qid):
+            return _qid
+
+class ModelSizeCalculator:
+    FLOAT_BITWIDTH = 32
```

Comment:
I prefer not to use hardcoded constants. There is a simple way how to get this value using a standard Python functionality:
```
import ctypes
FLOAT_BITWIDTH = ctypes.sizeof(ctypes.c_float) * 8
```

Reply:
Good input! @AlexKoff88

Reply:
Pushed.